agimus_demos_common/agimus_demos_common/launch_utils.py

```diff
@@ -0,0 +1,107 @@
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration, PathJoinSubstitution
+from launch_ros.substitutions import FindPackageShare
+
+
+def generate_default_franka_args() -> list[DeclareLaunchArgument]:
+    """Generates list of default arguments expected to be public interface of
+        launch files used by Agimus Demos for Franka robots.
+
+    Returns:
+        list[DeclareLaunchArgument]: List of DeclareLaunchArgument objects with
+            arguments expected by `franka_common.launch.py` and `franka_common_lfc.launch.py`
+    """
+    return [
+        DeclareLaunchArgument(
+            "robot_ip",
+            default_value="",
+            description="Hostname or IP address of the robot. "
+            + "If not empty launch file is configured for real robot. "
+            + "If empty `use_gazebo` is expected to be set to `true`.",
+        ),
+        DeclareLaunchArgument(
+            "arm_id",
+            default_value="fer",
+            description="ID of the type of arm used. Supported values: fer, fr3, fp3",
+            choices=["fer", "fr3", "fp3"],
+        ),
+        DeclareLaunchArgument(
+            "use_gazebo",
+            default_value="false",
+            description="Configures launch file for Gazebo simulation. "
+            + "If set to `true` launch file is configured for simulated robot. "
+            + "If set to `false` argument `robot_ip` is expected not to be empty.",
+            choices=["true", "false"],
+        ),
+        DeclareLaunchArgument(
+            "use_rviz",
+            default_value="false",
+            description="Visualize the robot in RViz",
+            choices=["true", "false"],
+        ),
+        DeclareLaunchArgument(
+            "gz_verbose",
+            default_value="false",
+            description="Whether to set verbosity level of Gazebo to 3.",
+            choices=["true", "false"],
+        ),
+        DeclareLaunchArgument(
+            "gz_headless",
+            default_value="false",
+            description="Whether to launch Gazebo in headless mode "
+            + "(no GUI is launched, only physics server).",
+            choices=["true", "false"],
+        ),
+    ]
+
+
+def generate_include_franka_launch(launch_file_name: str) -> IncludeLaunchDescription:
```

Comment:
I this necessary?
Is the code below too verbose?

Reply:
The iea behind it was that in case we decide to add new argument to `generate_default_franka_args()` we only update `generate_include_franka_launch()` and not all demos. Plus it makes the launch files shorter. Unfortunately the cost is that some parts of the launch file are hidden